### PR TITLE
Add service level settings to annotated services

### PR DESCRIPTION
**Pull request by sivaalli**
Motivation:

It is good to provide an ability to user to configure per service level settings for annotated service.

Closes: https://github.com/line/armeria/issues/2023

Modifications:

1. Pulled all methods that `AbstractServiceBindingBuilder`(minus AbstractBindingBuilder)  has into an interface called `ServiceBuilder`
2. `AbstractServiceBindingBuilder` implements this(ServiceBuilder) interface and also extends `AbstractBindingBuilder`
3. Provided a default implementation of `ServiceBuilder` called `DefaultServiceBuilder`, this is shared between `AbstractServiceBindingBuilder` and `AbstractServiceBuilder` to avoid code duplication

Result:
AnnotatedService can now be configured to have service specific properties.

Previously created PR: https://github.com/line/armeria/pull/2180. Sorry for closing and re-opening pr's 😞

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ * <pre>{@code
+ * ServerBuilder sb = Server.builder();
+ * sb.annotatedService()                                    // Returns an instance of this class
+ *   .requestTimeoutMillis(5000)
+ *   .maxRequestLength(8192)
+ *   .exceptionHandler((ctx, request, cause) -> HttpResponse.of(400))
+ *   .pathPrefix("/foo")
+ *   .verboseResponses(true)
+ *   .contentPreview(500)
+ *   .build(new Service())                                   // Return to the ServerBuilder.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java]: nit: The line comments (`// Return...`) could be moved left a little bit?

File: core/src/main/java/com/linecorp/armeria/server/DefaultServiceConfigSetters.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    @Override
+    public ServiceConfigSetters accessLogFormat(String accessLogFormat) {
+        return accessLogWriter(AccessLogWriter.custom(requireNonNull(accessLogFormat, "accessLogFormat")),
+                               true);
+    }
+
+    @Override
+    public ServiceConfigSetters accessLogWriter(AccessLogWriter accessLogWriter,
+                                          boolean shutdownOnStop) {
```
[the remark this answers is not in the document]
Comment (sivaalli) [reply core/src/main/java/com/linecorp/armeria/server/DefaultServiceConfigSetters.java]: Fixed. Please check again.

File: core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java (hunk 1 above)
Comment (sivaalli) [reply core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java]: Fixed. Please check again.

File: core/src/main/java/com/linecorp/armeria/server/DefaultServiceConfigSetters.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/server/DefaultServiceConfigSetters.java]: Thanks! Will merge once @ikhoon approves.

**Comment by sivaalli** [comment]
> Very nice work, @sivaalli ! Sorry it took longer than expected to review. 🙏

No problem, I can understand how busy you guys must be. :)

File: core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    @Override
+    public AnnotatedServiceBindingBuilder accessLogWriter(AccessLogWriter accessLogWriter,
+                                                          boolean shutdownOnStop) {
+        defaultServiceConfigSetters.accessLogWriter(accessLogWriter, shutdownOnStop);
+        return this;
+    }
+
+    /**
+     * Registers the given service to {@link ServerBuilder} adn return {@link ServerBuilder}
```
[the remark this answers is not in the document]
Comment (sivaalli) [reply core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java]: Oops, sorry. Fixed it. Please check again.

**Comment by trustin** [comment]
Boom! Thanks again, @sivaalli :heart:

**Pull request merged**